src/requested_path.rs

```diff
@@ -13,15 +14,27 @@ fn decode_percents(string: &&str) -> String {
     percent_decode(string.as_bytes()).decode_utf8().unwrap().into_owned()
 }
 
+fn normalize_path(path: &Path) -> PathBuf {
+    path.components().fold(PathBuf::new(), |mut result, p| {
+        match p {
+            Component::Normal(x) => {
+                result.push(x);
+                result
+            }
+            Component::ParentDir => {
+                result.pop();
+                result
+            },
+            _ => result
+        }
+    })
+}
+
 impl RequestedPath {
     pub fn new<P: AsRef<Path>>(root_path: P, request: &Request) -> RequestedPath {
+        let decoded_req_path = PathBuf::from_iter(request.url.path().iter().map(decode_percents));
```

Comment:
Doesn't `Path.canonicalize()` touch an underlying filesystem, calling systemcall? If we don't have a permission to a intermediate directory, I think it would fail. Is that okay?

This PR's approach is inspired by Python's simple builtin httpserver, available via `python3 -m http.server`
See https://github.com/python/cpython/blob/c30098c8c6014f3340a369a31df9c74bdbacc269/Lib/http/server.py#L756